app/src/main/java/org/wikipedia/analytics/eventplatform/EventPlatformClient.java

```diff
@@ -0,0 +1,451 @@
+package org.wikipedia.analytics.eventplatform;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Timer;
+import java.util.TimerTask;
+
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.deleteStoredSessionId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.fetchStreamConfigs;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getAppInstallId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getIso8601Timestamp;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getStoredSessionId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.getStoredStreamConfigs;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.isOnline;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.postEvents;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.setStoredSessionId;
+import static org.wikipedia.analytics.eventplatform.EventPlatformClientIntegration.setStoredStreamConfigs;
+import static org.wikipedia.analytics.eventplatform.SamplingConfig.Identifier.DEVICE;
+import static org.wikipedia.analytics.eventplatform.SamplingConfig.Identifier.PAGEVIEW;
+import static org.wikipedia.analytics.eventplatform.SamplingConfig.Identifier.SESSION;
+import static org.wikipedia.settings.Prefs.isEventLoggingEnabled;
+
+public final class EventPlatformClient {
```

Comment:
@sharvaniharan  Here's at least one possible explanation for test failures: Mockito is not able to mock `final` classes.